src/main/java/com/manardenza/dao/AbstractDao.java

```diff
@@ -2,27 +2,60 @@
 
 import com.manardenza.entity.AbstractObject;
 
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
 import java.util.ArrayList;
 import java.util.List;
 
 public abstract class AbstractDao<T extends AbstractObject> {
 
     protected List<T> database = new ArrayList<>();
+    protected File databaseFile;
+
+    private void saveToFile() {
+        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(databaseFile))) {
+            oos.writeObject(database);
+        } catch (IOException e) { //FIXME: Either log or rethrow this exception.
+            e.printStackTrace();//FIXME: Use a logger to log this exception.
+        }
+    }
+
+    private void readFromFile() {
+        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(databaseFile))) {
+            database = (List<T>) ois.readObject();
+        } catch (IOException | ClassNotFoundException e) { //FIXME: Either log or rethrow this exception.
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)